src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -522,6 +574,98 @@ public function findOr($id, $columns = ['*'], Closure $callback = null)
         return $callback();
     }
 
+    /**
+     * Find a model by its primary key or load by its identity.
+     *
+     * @param  mixed  $id
+     * @param  array|string  $columns
+     * @return \Illuminate\Database\Eloquent\Model|\Illuminate\Database\Eloquent\Collection|static[]|static|null
+     */
+    public function findOrIdentify($id, $columns = ['*'])
+    {
+        if (is_array($id) || $id instanceof Arrayable) {
+            return $this->findMany($id, $columns);
+        }
+
+        return $this->identifyModel($id) ?? $this->whereKey($id)->first($columns);
+    }
+
+    /**
+     * Get the identity manager.
+     *
+     * @return \Illuminate\Database\Eloquent\IdentityManager
+     */
+    public function getIdentityManager()
+    {
+        if (! isset($this->identityManager)) {
+            $this->identityManager = app(IdentityManager::class);
```

Comment:
I **think** that this `app` call makes `illuminate/database` impossible to ship as a standalone package now, because using it here basically couples it to `laravel/framework`. 

Wouldn't it be better to set it as a static variable within a `ServiceProvider`, just like the `Dispatcher`, `Connection` etc. on `Model` instances? This way, the coupling can be eliminated.